src/librustdoc/html/static/js/search.js

```diff
@@ -1106,6 +1106,137 @@ class RoaringBitmapBits {
     }
 }
 
+/**
+ * @typedef {{
+ *     children: [NameTrie],
+ *     matches: [number],
+ * }}
+ */
+class NameTrie {
+    constructor() {
+        this.children = [];
+        this.matches = [];
+    }
+    insert(name, id, tailTable) {
+        this.insertSubstring(name, 0, id, tailTable);
+    }
+    insertSubstring(name, substart, id, tailTable) {
+        const l = name.length;
+        if (substart === l) {
+            this.matches.push(id);
+        } else {
+            const sb = name.charCodeAt(substart);
+            let child;
+            if (this.children[sb] !== undefined) {
```

Comment:
Wouldn't it be better to check `this.children.length < sb`?

Reply:
This is a [sparse array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays).

Reply:
Please add this explanation and link on the field definition please. :)

Reply:
Okay, it's added.